### PR TITLE
MAINT: Adjust Pull-Request labeler configuration

### PR DESCRIPTION
**Pull request by jjerphan**
#### Reference issue

Follow-up of #16870

#### What does this implement/fix?

This adapts the labeler configuration to resolve issues exposed in https://github.com/scipy/scipy/pull/16870#issuecomment-1251301776.

#### Additional information

This is still WIP.

cc @tupui

### CODE REVIEW

File: .github/workflows/pull-request-labeler.yml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-    types: [created]
+    types: [opened]
 
 jobs:
 
   label_pull_request:
     runs-on: ubuntu-latest
     steps:
-    - uses: thomasjpfan/labeler@v2.5.0
+    - uses: actions/labeler@v4.0.1
```
Comment (tupui) [inline .github/workflows/pull-request-labeler.yml:11]: Please test this on your fork. Last time we had this action, the sync mechanism was not working. The issue is still open with a workaround I did not test https://github.com/actions/labeler/pull/113

**Review by tupui — COMMENTED** [review]
Thanks @jjerphan for following up on this 😃

File: .github/workflows/pull-request-labeler.yml (hunk 1 above)
Comment (jjerphan) [reply .github/workflows/pull-request-labeler.yml]: I am comming back to this PR after an eternity: I plan to maintain the changes of 
https://github.com/actions/labeler/pull/113 on my fork in the meantime. I still need to assert whether it works or not. If it does, would depending on this fork work for you or would you prefer depending on https://github.com/actions/labeler/ solely?

**Comment by jjerphan** [comment]
I am coming back to this PR after an eternity: I plan to maintain the changes of 
https://github.com/actions/labeler/pull/113 on my fork in the meantime. I still need to assert whether it works or not. If it does, would depending on this fork work for you or would you prefer depending on [`actions/labeler`](https://github.com/actions/labeler/) solely?

**Comment by tupui** [comment]
@jjerphan sorry for the delay, I was away.

If the fix in the reported issue works with the original repo, that would be better. We would prefer not to depend on custom actions (extra work to maintain) unless they are used by other large projects.

**Comment by jjerphan** [comment]
> @jjerphan sorry for the delay, I was away.

It's fine. Congrats for [getting a CZI grant to pursue your work on Scipy](https://chanzuckerberg.com/eoss/proposals/scipy-fundamental-tools-for-biomedical-research/)! :partying_face:

**Comment by tupui** [comment]
Thanks!

**Comment by jjerphan** [comment]
Hi @tupui, I am coming back to you after another eternity.

https://github.com/actions/labeler/pull/480 which pursues https://github.com/actions/labeler/pull/113 is on the verge to be approved. I think the best is that we wait for it to be merged before taking further decisions.

**Comment by tupui** [comment]
Thanks! Let's do that. Seems like a new team is taking this over, great news.

**Review by tupui — APPROVED** [review]
LGTM, seen this in-person. In it goes, thanks Julien 😃

**Comment by jjerphan** [comment]
See those PRs to check behavior with the setup on my fork:
 - https://github.com/jjerphan/scipy/pull/6
 - https://github.com/jjerphan/scipy/pull/5

**Comment by jjerphan** [comment]
Hi @j-bowhay,

I have seen that you have been triaging issues and PRs lately.

I am interested to know what you think of the setup this CI brought to see if this can be refined.

**Comment by j-bowhay** [comment]
I think it is working well. One possible point for further improvement would be to pick off the acronym (eg. MAINT) and map this to a label as at the moment this still has to be done manually.

**Pull request merged**